server/render/index.js

```diff
@@ -86,8 +86,7 @@ export function serverRender( req, res ) {
 		config.isEnabled( 'server-side-rendering' ) &&
 		context.layout &&
 		! context.user &&
-		isEmpty( context.query ) &&
```

Comment:
Yes, this worries me. Aside from crawlers slowing the loop down, we could conceivably attack the server to DoS it with ease.